swap/src/asb/event_loop.rs

```diff
@@ -319,13 +320,40 @@ where
         min_buy: bitcoin::Amount,
         max_buy: bitcoin::Amount,
     ) -> Result<BidQuote> {
+        let monero_decimal = self.monero_wallet.get_balance().await?.as_monero_decimal();
+
         let rate = self
             .latest_rate
             .latest_rate()
-            .context("Failed to get latest rate")?;
+            .context("Failed to get latest rate")?
+            .ask()
+            .context("Failed to compute asking price")?;
+
+        let bitcoin_decimal: Decimal = Decimal::from(rate.as_sat()) / Decimal::from(100_000_000);
+
+        let max_bitcoin_for_monero = match (monero_decimal * bitcoin_decimal).to_f64() {
+            Some(val) => bitcoin::Amount::from_btc(val)?,
```

Comment:
Use `to_u64` instead and `from_sat` please. `from_btc` involves floating point arithmetic!

Reply:
I managed a way to use u64 but had to watch for an underflow for subtracting the Monero fee